server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/newyorkcoin/NewyorkcoinExtension.java

```diff
@@ -0,0 +1,122 @@
+        try {
+        if (walletLogin !=null && !walletLogin.trim().isEmpty()) {
+            StringTokenizer st = new StringTokenizer(walletLogin,":");
+            String walletType = st.nextToken();
+
+            if ("newyorkcoind".equalsIgnoreCase(walletType)
+                || "newyorkcoindnoforward".equalsIgnoreCase(walletType)) {
+                //"newyorkcoind:protocol:user:password:ip:port:accountname"
+
+                String protocol = st.nextToken();
+                String username = st.nextToken();
+                String password = st.nextToken();
+                String hostname = st.nextToken();
+                int port = Integer.parseInt(st.nextToken());
+                String label = "";
+                if (st.hasMoreTokens()) {
+                    label = st.nextToken();
+                }
+
+                InetSocketAddress tunnelAddress = ctx.getTunnelManager().connectIfNeeded(walletLogin, tunnelPassword, InetSocketAddress.createUnresolved(hostname, port));
+                hostname = tunnelAddress.getHostString();
+                port = tunnelAddress.getPort();
+
+                if (protocol != null && username != null && password != null && hostname !=null && label != null) {
+                    String rpcURL = protocol +"://" + username +":" + password + "@" + hostname +":" + port;
+                    if ("newyorkcoindnoforward".equalsIgnoreCase(walletType)) {
+                        return new NewyorkcoindUniqueAddressRPCWallet(rpcURL);
+                    }
+                    return new NewyorkcoindRPCWallet(rpcURL, label);
+                }
+            }
+        }
+        } catch (Exception e) {
+            log.error("", e);
+        }
+        return null;
+    }
+
+    @Override
+    public ICryptoAddressValidator createAddressValidator(String cryptoCurrency) {
+        if (CryptoCurrency.NYC.getCode().equalsIgnoreCase(cryptoCurrency)) {
+            return new NewyorkcoinAddressValidator();
+        }
+        return null;
+    }
+
+    @Override
+    public IRateSource createRateSource(String sourceLogin) {
+        if (sourceLogin != null && !sourceLogin.trim().isEmpty()) {
+            StringTokenizer st = new StringTokenizer(sourceLogin,":");
+            String rsType = st.nextToken();
+
+            
+            } else if ("nycfix".equalsIgnoreCase(exchangeType)) {
+                BigDecimal rate = BigDecimal.ZERO;
+                if (st.hasMoreTokens()) {
+                    try {
+                        rate = new BigDecimal(st.nextToken());
+                    } catch (Throwable e) {
+                    }
+                }
+                String preferedFiatCurrency = FiatCurrency.USD.getCode();
+                if (st.hasMoreTokens()) {
+                    preferedFiatCurrency = st.nextToken().toUpperCase();
+                }
+                return new FixPriceRateSource(rate,preferedFiatCurrency);
+
+            }
```

Comment:
Remove this extra closing curly bracket and format your code + remove empty lines from if block